Index: lib/active_record/fixtures.rb
===================================================================
--- lib/active_record/fixtures.rb  (revision 4267)
+++ lib/active_record/fixtures.rb  (working copy)
@@ -394,15 +394,20 @@
   def value_list
     @fixture.values.map { |v| ActiveRecord::Base.connection.quote(v).gsub('\\n', "\n").gsub('\\r', "\r") }.join(", ")
   end
-
+  
+  def klass
+    @class_name.is_a?(Class) ? @class_name : Object.const_get(@class_name)
+  rescue
+    raise FixtureClassNotFound, "The class #{@class_name.inspect} was not found."
+  end
+  
   def find
-    klass = @class_name.is_a?(Class) ? @class_name : Object.const_get(@class_name) rescue nil
-    if klass
-      klass.find(self[klass.primary_key])
-    else
-      raise FixtureClassNotFound, "The class #{@class_name.inspect} was not found."
-    end
+    klass.find(find_id)
   end
+  
+  def find_id
+    self[klass.primary_key]
+  end
 
   private
     def read_fixture_file(fixture_file_path)
@@ -480,6 +485,14 @@
               raise StandardError, "No fixture with name '#{fixture}' found for table '#{table_name}'"
             end
           end
+          
+          define_method(table_name + "_id") do |fixture|
+            if @loaded_fixtures[table_name][fixture.to_s]
+              @loaded_fixtures[table_name][fixture.to_s].find_id
+            else
+              raise StandardError, "No fixture with name '#{fixture}' found for table '#{table_name}'"
+            end
+          end
         end
       end
 
